DOCUMENTATION.md

```diff
@@ -1370,529 +1395,202 @@ The following are example configurations.
 
-An example configuration is shown below.
+### 7.3.2. Handling Retry-able Operation Failure
 
-```json
-{
-    "network": {
-        "ssl": {
-            "enabled": true,
-            "factory": {
-                "path": "my_factory.js",
-                "exportedName": "SSLFactory",
-                "properties": {
-                    "caPath": "ca.pem",
-                    "keyPath": "key.pem",
-                    "certPath": "cert.pem",
-                    "keepOrder": true
-                }
-            }
-        }
-    }
-}
-```
+While sending the requests to the related members, the operations can fail due to various reasons. Read-only operations are retried by default. If you want to enable retrying for the other operations, you can set the `redoOperation` to `true`. See the [Enabling Redo Operation section](#53-enabling-redo-operation).
 
+You can set a timeout for retrying the operations sent to a member. This can be provided by using the property `hazelcast.client.invocation.timeout.seconds` in `ClientConfig.properties`. The client will retry an operation within this given period, of course, if it is a read-only operation or you enabled the `redoOperation` as stated in the above paragraph. This timeout value is important when there is a failure resulted by either of the following causes:
 
-An example of a factory, `my_factory.js`, is shown below.
+* Member throws an exception.
+* Connection between the client and member is closed.
+* Client’s heartbeat requests are timed out.
 
+When a connection problem occurs, an operation is retried if it is certain that it has not run on the member yet or if it is idempotent such as a read-only operation, i.e., retrying does not have a side effect. If it is not certain whether the operation has run on the member, then the non-idempotent operations are not retried. However, as explained in the first paragraph of this section, you can force all the client operations to be retried (`redoOperation`) when there is a connection failure between the client and member. But in this case, you should know that some operations may run multiple times causing conflicts. For example, assume that your client sent a `queue.offer` operation to the member and then the connection is lost. Since there will be no response for this operation, you will not know whether it has run on the member or not. If you enabled `redoOperation`, it means this operation may run again, which may cause two instances of the same object in the queue.
 
-```javascript
-function SSLFactory() {
-}
+## 7.4. Using Distributed Data Structures
 
-SSLFactory.prototype.init = function (properties) {
-    var promises = [];
-    var readFile = Promise.promisify(fs.readFile);
-    this.keepOrder = props.userDefinedProperty1;
-    var self = this;
+Most of the distributed data structures are supported by the Node.js client. In this chapter, you will learn how to use these distributed data structures.
```

Comment:
nit: change to `Most of the distributed data structures available in IMDG are supported by the Node.js client.`